### PR TITLE
feat (api): Adds ability for click insights events to be triggered

### PR DESCRIPTION
**Pull request by 8bittitan**
Recently we added insights options for DocSearch in #1904. This PR now also adds in the ability for click analytics to be triggered as well when insights option is active.

### CODE REVIEW

File: packages/docsearch-react/src/utils/buildInsightsClickParams.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  ClickedObjectIDsAfterSearchParams,
+} from '@algolia/autocomplete-plugin-algolia-insights';
+
+import type { InternalDocSearchHit } from '../types';
+
+export function buildInsightsClickParams(
+  item: InternalDocSearchHit,
+  itemPosition: number
+): InsightsParamsWithItems<ClickedObjectIDsAfterSearchParams> {
+  const insightsItem = item as AlgoliaInsightsHit;
```
Comment (levimichael) [inline packages/docsearch-react/src/utils/buildInsightsClickParams.ts:13]: Could we run into issues with this assertion?

**Review by levimichael — APPROVED** [review]
Cool! 🥳

File: packages/docsearch-react/src/utils/buildInsightsClickParams.ts (hunk 1 above)
Comment (8bittitan) [reply packages/docsearch-react/src/utils/buildInsightsClickParams.ts:13]: I don't think so, if we have an `insights` plugin active the needed attributes are added to the `item`. All `AlgoliaInsightsHit` type adds is better typings for the `__autocomplete_*` attributes.

File: packages/docsearch-react/src/Hit.tsx [hunk 2]
```diff
@@ -8,5 +8,9 @@ interface HitProps {
 }
 
 export function Hit({ hit, children }: HitProps) {
-  return <a href={hit.url}>{children}</a>;
+  return (
+    <a href={hit.url} onClick={(e) => e.preventDefault()}>
```
[the remark this answers is not in the document]
Comment (8bittitan) [reply packages/docsearch-react/src/Hit.tsx]: Crap, that was for debugging 😅 Removed :D

**Review by shaneafsar — APPROVED** [review]

File: packages/docsearch-react/src/utils/buildInsightsClickParams.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import type { AlgoliaInsightsHit } from '@algolia/autocomplete-core';
+import type {
+  InsightsParamsWithItems,
+  ClickedObjectIDsAfterSearchParams,
+} from '@algolia/autocomplete-plugin-algolia-insights';
+
+import type { InternalDocSearchHit } from '../types';
+
+export function buildInsightsClickParams(
```
Comment (francoischalifour) [inline packages/docsearch-react/src/utils/buildInsightsClickParams.ts:9]: It looks like this function content can be inlined in `sendItemClickEvent`?

File: packages/docsearch-react/src/DocSearchModal.tsx [hunk 4]
```diff
@@ -130,6 +131,23 @@ export function DocSearchModal({
...
+      const insightsClickParams = buildInsightsClickParams(
+        item,
+        item.__autocomplete_id
+      );
+
+      state.context.algoliaInsightsPlugin.insights.clickedObjectIDsAfterSearch(
+        insightsClickParams
+      );
+    },
+    [state]
```
Comment (francoischalifour) [inline packages/docsearch-react/src/DocSearchModal.tsx:148]: Watching for plugin changes would be more accurate.

```suggestion
    [state.context.algoliaInsightsPlugin]
```

File: packages/docsearch-react/src/utils/buildInsightsClickParams.ts (hunk 3 above)
Comment (8bittitan) [reply packages/docsearch-react/src/utils/buildInsightsClickParams.ts:9]: Yeah I agree, originally I had the insights inlined to the click handler so this felt better then. Now that it's broken out this makes sense

File: packages/docsearch-react/src/DocSearchModal.tsx (hunk 4 above)
Comment (8bittitan) [reply packages/docsearch-react/src/DocSearchModal.tsx:148]: Good call

File: packages/docsearch-react/src/Hit.tsx (hunk 2 above)
Comment (francoischalifour) [inline packages/docsearch-react/src/Hit.tsx:12]: Looks like the  `preventDefault` call should be removed again.

**Pull request merged**